pkg/services/publicdashboards/publicdashboard.go

```diff
@@ -26,7 +26,7 @@ type Service interface {
 	Find(ctx context.Context, uid string) (*PublicDashboard, error)
 	Create(ctx context.Context, u *user.SignedInUser, dto *SavePublicDashboardDTO) (*PublicDashboard, error)
 	Update(ctx context.Context, u *user.SignedInUser, dto *SavePublicDashboardDTO) (*PublicDashboard, error)
-	Delete(ctx context.Context, uid string) error
```

Comment:
Because the interface changed, the pipeline is failing. Can you regenerate the service mock file using [mockery](https://github.com/grafana/grafana/blob/main/pkg/services/publicdashboards/publicdashboard.go#L16) and push your changes to fix the build?

Reply:
hey, I am getting this error on running `go generate.` Can you help? `flag provided but not defined: -structname`

Reply:
Are you running this command in the same folder?
```
mockery --name Service --structname FakePublicDashboardService --inpackage --filename public_dashboard_service_mock.go
```

Reply:
yes

Reply:
It seems to be an error from Go. Did you install mockery? You can try running it without `go generate` or adding flags to view the logs.

Reply:
yes, I installed mockery. I tried running it without` go generate` as well. Do let me know. It doesn't seem to work

Reply:
Ok, you can add changes manually, they are pretty simple, just adding the new parameter in the `Delete` method definition in the `public_dashboard_service_mock.go` file:
```
// Delete provides a mock function with given fields: ctx, uid, dashboardUid
func (_m *FakePublicDashboardService) Delete(ctx context.Context, uid string, dashboardUid string) error {
	ret := _m.Called(ctx, uid, dashboardUid)

	var r0 error
	if rf, ok := ret.Get(0).(func(context.Context, string, string) error); ok {
		r0 = rf(ctx, uid, dashboardUid)
	} else {
		r0 = ret.Error(0)
	}

	return r0
}
```